MQTT_GW: Dynamic subscribe topic declaration failed

I'm trying to dynamically declare the variables MY_MQTT_PUBLISH_TOPIC_PREFIX and MY_MQTT_SUBSCRIBE_TOPIC_PREFIX with de following code:

```
char mqtt_publish_topic[34];
char mqtt_subscribe_topic[33];
#define MY_MQTT_PUBLISH_TOPIC_PREFIX mqtt_publish_topic
#define MY_MQTT_SUBSCRIBE_TOPIC_PREFIX mqtt_subscribe_topic
```

I get the this compilation error:

```
In file included from <XXX>\Arduino\libraries\MySensors/MySensors.h:221:0,
                 from <XXX>\GatewayESP8266MQTTClient.ino:115:
<XXX>\Arduino\libraries\MySensors/core/MyGatewayTransportMQTTClient.cpp: In function 'bool reconnectMQTT()':
<XXX>\Arduino\libraries\MySensors/core/MyGatewayTransportMQTTClient.cpp:148:58: error: expected ')' before string constant
    _MQTT_client.subscribe(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX "/+/+/+/+/+");
```

If I remove the ___builtin_constant_p(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX)_ verification code in _MyGatewayTransportMQTTClient.cpp_, changing

```
// Once connected, publish subscribe
if (__builtin_constant_p(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX)) {
	// to save some memory
	_MQTT_client.subscribe(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX "/+/+/+/+/+");
} else {
	char inTopic[strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + strlen("/+/+/+/+/+")];
	(void)strncpy(inTopic, MY_MQTT_SUBSCRIBE_TOPIC_PREFIX, strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + 1);
	(void)strcat(inTopic, "/+/+/+/+/+");
	_MQTT_client.subscribe(inTopic);
}
```

by

```
char inTopic[strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + strlen("/+/+/+/+/+")];
(void)strncpy(inTopic, MY_MQTT_SUBSCRIBE_TOPIC_PREFIX, strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + 1);
(void)strcat(inTopic, "/+/+/+/+/+");
_MQTT_client.subscribe(inTopic);
```

It's work fine.

If the variable declaration must be done differently, please tell me how i can do it, or I can do a pull request with this modificacion if you want.

Related: #789
Related: #1240

--- a/core/MyGatewayTransportMQTTClient.cpp
+++ b/core/MyGatewayTransportMQTTClient.cpp
@@ -144,15 +144,10 @@ bool reconnectMQTT(void)
 		// Send presentation of locally attached sensors (and node if applicable)
 		presentNode();
 		// Once connected, publish subscribe
-		if (__builtin_constant_p(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX)) {
-			// to save some memory
-			_MQTT_client.subscribe(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX "/+/+/+/+/+");
-		} else {
-			char inTopic[strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + strlen("/+/+/+/+/+")];
-			(void)strncpy(inTopic, MY_MQTT_SUBSCRIBE_TOPIC_PREFIX, strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + 1);
-			(void)strcat(inTopic, "/+/+/+/+/+");
-			_MQTT_client.subscribe(inTopic);
-		}
+		char inTopic[strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + strlen("/+/+/+/+/+")];
+		(void)strncpy(inTopic, MY_MQTT_SUBSCRIBE_TOPIC_PREFIX, strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + 1);
+		(void)strcat(inTopic, "/+/+/+/+/+");
+		_MQTT_client.subscribe(inTopic);
 
 		return true;
 	}

